saq/worker.py

```diff
@@ -240,6 +236,20 @@ def _process(self, previous_task=None):
             new_task.add_done_callback(self._process)
 
 
+def ensure_coroutine_function(func):
+    if asyncio.iscoroutinefunction(func):
+        return func
+
+    @wraps(func)
```

Comment:
possible to simplify this? do we need wraps because we don't use a decorator, also partial?

async def wrapped(*args, **kwargs):
     return await loop.run_in_executor(executor=None, func=lambda: func(*args, **kwargs)
 return wrapped

Reply:
Ah sure, I can switch to lambda. 

This is a decorator, it's just not invoked with the `@` syntax. But I guess that means its unlikely someone will have a direct handle on the wrapped function, so I'll remove it.